docs/src/explanations/connectable.md

```diff
@@ -0,0 +1,418 @@
+
+For connections where you want to every producer field to always drive every consumer field, regardless of alignment, use `:#=` (the "coercion connection"). This operator is useful for initializing wires whose types contain members of mixed alignment.
+
+```scala mdoc:silent
+import chisel3.experimental.BundleLiterals._
+class Example4 extends RawModule {
+  val bar = Wire(new MixedAlignmentBundle)
+  dontTouch(bar) // So we see it in the output verilog
+  bar :#= (new MixedAlignmentBundle).Lit(_.a -> true.B, _.b -> true.B)
+}
+```
+
+This generates the following Verilog, where all members are driven `foo` to `bar`, regardless of alignment:
+
+```scala mdoc:verilog
+import chisel3.stage.ChiselStage
+
+ChiselStage.emitVerilog(new Example4())
+```
+
+## Waived Data
+
+It is not uncommon for a user to want to connect Chisel components which are not type equivalent. For example, a user may want to cook up the `ready`/`valid` fields of a `ReadyValidIO` to a `DecoupledIO`, but
+because the `bits` field is not present in both, our operators would reject a connection.
+
+`WaivedData` is the mechanism to specialize connection operator behavior in these scenarios. For any addition field which is not present in the other component being connected to, they can be explicitly waived from the operator to be ignored, rather than trigger an error.
+
+### Connecting sub-types to super-types by waiving extra fields
+
+In the following example, we can use `:<>=` to connect a `MyReadyValid` to a `MyDecoupled` by waiving the `bits` member.
+
+```scala mdoc:silent
+class MyReadyValid extends Bundle {
+  val valid = Bool()
+  val ready = Flipped(Bool())
+}
+class MyDecoupled extends MyReadyValid {
+  val bits = UInt(32.W)
+}
+class Example5 extends RawModule {
+  val in  = IO(Flipped(new MyDecoupled()))
+  val out = IO(new MyReadyValid())
+  out :<>= in.waive(_.bits)
+}
+```
+
+This generates the following Verilog, where `ready` and `valid` are connected, and `bits` is ignored:
+
+```scala mdoc:verilog
+import chisel3.stage.ChiselStage
+
+ChiselStage.emitVerilog(new Example5())
+```
+
+### Connecting types with optional fields
+
+In the following example, we can use `:<>=` and `waive` to connect two `MyDecoupledOpts`'s, where only one has a `bits` member.
+
+```scala mdoc:silent
+class MyDecoupledOpt(hasBits: Boolean) extends Bundle {
+  val valid = Bool()
+  val ready = Flipped(Bool())
+  val bits = if (hasBits) Some(UInt(32.W)) else None
+}
+class Example6 extends RawModule {
+  val in  = IO(Flipped(new MyDecoupledOpt(true)))
+  val out = IO(new MyDecoupledOpt(false))
+  out :<>= in.waive(_.bits.get) // We can know to call .get because we can inspect in.bits.isEmpty
+}
+```
+
+This generates the following Verilog, where `ready` and `valid` are connected, and `bits` is ignored:
+
+```scala mdoc:verilog
+import chisel3.stage.ChiselStage
+
+ChiselStage.emitVerilog(new Example6())
+```
+
+### Connecting different sub-types
```

Comment:
maybe before waivable we should explain the different sub-types assignment anyway, like:

```
class HasBits extends MyReadyValid {
  val bits = UInt(32.W)
}
```
```
class ADifferentHasBits extends MyReadyValid {
  val bits = UInt(32.W)
}
```

show how to cast them both to MyReadyValid -- but importantly they WILL still connect their `bits`! As opposed to .viewAsSuperType[MyReadyValid]!
This is important distinction